internal/server/config.go

```diff
@@ -603,14 +604,18 @@ func (s Server) loadConfig(config Config) error {
 	if err := validate.Validate(config); err != nil {
 		return err
 	}
-	org, err := data.GetOrganization(s.DB(), data.ByName(models.DefaultOrganizationName))
-	if err != nil {
-		return err
-	}
 
+	org := s.db.DefaultOrg
 	return s.db.Transaction(func(db *gorm.DB) error {
 		tx := data.NewTransaction(db, org.ID)
 
+		if config.DefaultOrganizationDomain != org.Domain {
```

Comment:
Kind of, today the behaviour is:
* when EnableSignup=false we ignore baseDomain entirely. If no org is specified, use the default org
* when EnableSignup=true we only use base domain for signup. If no org is specified, then error if the endpoint requires an org

This PR doesn't change that behaviour, but it does mean that anyone using `EnableSignup=true` should also set `DefaultOrganizationDomain`, otherwise they likely won't be able to login.